chain/chain/src/chain.rs

```diff
@@ -936,6 +941,17 @@ impl Chain {
             .into());
         }
 
+        let ShardStateSyncResponse {
+            chunk,
+            chunk_proof,
+            prev_chunk_header,
+            prev_chunk_proof,
+            prev_payload,
+            block_transactions,
+            incoming_receipts_proofs,
+            root_proofs,
+        } = shard_state;
+
         // 1. Checking that chunk header is at least valid
         // 1a. Checking chunk.header.hash
         if chunk.header.hash != ChunkHash(hash(&chunk.header.inner.try_to_vec()?)) {
```

Comment:
I don't think we need this check, `hash` is computed locally, so it must match the hash of the chunk.